Chapter05/Stack_using_array.py

```diff
@@ -1,41 +1,40 @@
 size = 3
-data = [0]*(size)   #Initialize the stack
+data = [0] * (size)  # Initialize the stack
 top = -1
 
+
 def push(x):
-  global top
-  if top >= size-1:
-    print("Stack Overflow")
-  else:
-    top = top + 1
-    data[top] = x
+    global top
+    if top >= size - 1:
+        print("Stack Overflow")
+    else:
+        top = top + 1
+        data[top] = x
 
 
 def pop():
-  global top
-  if top == -1:
-    print("Stack Underflow")
-  else:
-    top = top-1
-    data[top] = 0
-    return data[top+1]
+    global top
+    if top == -1:
+        print("Stack Underflow")
+    else:
+        value = data[top]
+        data[top] = 0
+        top = top - 1
+        return value
 
 
 def peek():
-  global top
```

Comment:
Since we do not assign to the `top`, there is no need for `global` keyword here.

Reply:
Yes. Why there are so many errors in this book?

Reply:
I tried to contact the author for reviewing the PRs but he didn't respond to my email.